### PR TITLE
Add codegen snippets for "Provider" Page

### PR DESCRIPTION
**Pull request by leehack**
Added codegen snippets for "Provider" Page.

### CODE REVIEW

**Comment by leehack** [comment]
@rrousselGit Let me know if it's OK. If it's good, I can continue for more pages.

**Comment by rrousselGit** [comment]
Could you fix the CI? You likely created some broken URLs and possible analysis errors.

**Comment by leehack** [comment]
> Could you fix the CI? You likely created some broken URLs and possible analysis errors.

Sure.

**Comment by leehack** [comment]
I had to bump the minimum SDK version to 2.13.0 for the doc project.
2.12.0 doesn't understand this.
```
typedef CompletedTodosRef = AutoDisposeProviderRef<List<Todo>>;
```

**Comment by rrousselGit** [comment]
The SDK bump is fine. It could/should be using latest.

But the CI is still failing. One of your imports or links is most likely incorrect

**Comment by leehack** [comment]
> The SDK bump is fine. It could/should be using latest.
> 
> But the CI is still failing. One of your imports or links is most likely incorrect

I can bump it to the latest but it complains about issues from other parts that I didn't modify. I don't know what's your preference. Do you prefer to include those changes in this PR or cut another PR?

**Comment by rrousselGit** [comment]
You don't need to bump the SDK, it's fine.
Just fix the CI :)

**Comment by leehack** [comment]
Cool. I am working on it. I'm not sure how I can locally test the ci/cd jobs properly. I may probably push a few times to make sure everything fine in ci/cd

**Comment by leehack** [comment]
@rrousselGit It seems the coverage fails for the part I did touch. Would it be a blocker to merge it into the docs-v2 branch?

**Comment by rrousselGit** [comment]
Ignore coverage. It's likely because you're not up-to-date with master

It's probably good to merge. But I'll have a deeper look at this later as I haven't executed it yet.

**Comment by leehack** [comment]
Yep. I just branched it from the `docs-v2`. Feel free to let me know if there's anything wrong. There might be mistakes since it's my first time contributing to this doc.

File: website/docs/providers/provider/todos_consumer.dart [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
-  // TODO show the todos using a ListView/GridView/.../* SKIP */
-  return Container();
-  /* SKIP END */
-});
+      Consumer(builder: (context, ref, child) {
+    final completedTodos = ref.watch(completedTodosProvider);
+    // TODO show the todos using a ListView/GridView/.../* SKIP */
+    return Container();
+    /* SKIP END */
+  });
```
Comment (rrousselGit) [inline website/docs/providers/provider/todos_consumer.dart:16]:
```suggestion
Consumer(builder: (context, ref, child) {
  final completedTodos = ref.watch(completedTodosProvider);
  // TODO show the todos using a ListView/GridView/.../* SKIP */
  return Container();
  /* SKIP END */
});
```

**Comment by rrousselGit** [comment]
Sorry I forgot to review this. LGTM!

**Pull request merged**